lib/rules/disallow-keywords-in-comments.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * Disallows keywords in your comments, such as TODO or FIXME
+ *
+ * Type: `Boolean` or `String` or `Array`
+ *
+ * Values:
+ * - `true`
+ * - `'\b(word1|word2)\b'`
+ * - `['word1', 'word2']`
+ *
+ * #### Examples
+ *
+ * ```js
+ * "disallowKeywordsInComments": true
+ * "disallowKeywordsInComments": "\\b(word1|word2)\\b"
+ * "disallowKeywordsInComments": ["word1", "word2"]
+ * ```
+ *
+ * #### Invalid:
+ * ```
+ * // ToDo
+ * //TODO
+ * /** fixme *\/
+ * /**
+ *  * FIXME
+ *  *\/
+ * ```
+ */
+
+var assert = require('assert');
+
+var getCommentErrors = function(comment, keywordRegEx) {
+    var splitComment = comment.value.split('\n');
+    var errors = [];
+
+    splitComment.forEach(function(commentNode, index) {
+        var lineIndex = index;
+        var matches = commentNode.match(keywordRegEx);
+        var lastIndex = 0;
+
+        if (!matches) { return; }
+
+        errors = errors.concat(matches.map(function(match) {
+            var foundAtIndex;
+
+            lastIndex = lastIndex === 0 ? lastIndex : lastIndex + 1;
+            foundAtIndex = commentNode.indexOf(match, lastIndex);
```

Comment:
No need for `foundAtIndex`, just assign directly to `lastIndex` and use it everywhere you used `foundAtIndex` below.